issue #1 and #2

zhaoli1211 opened the pull request:
More Pushed into Library
fix typo

zhaoli1211 committed:
More Pushed into Library
fix typo

plflanagan reviewed (commented):
Looks great. I was confused by `var textData: UNUMAttributeStringStruct?` though. Essentially this is for saving the attributes on our server, right? It doesn't seem like it really has any use in `ViewController`. So really it shouldn't be returned in `didChangeTextAttribute` or stored or sent anywhere in `ViewController`. the TextEditor should take an attributedString, and do the conversion itself if it needs to, but `ViewController` shouldn't have to know about it.

If it is needed somewhere outside the library, then I'd recommend the library just having a conversion function that takes an attributed string and returns the struct version.

Beyond that, though, I think this is spot on -- at least until we actually go to implement it, at which point we'll probably notice a few other things to change.

zhaoli1211 committed:
removed NSAttributeStringStruct
add NSAttributedString extension to get and set each attribute inside and outside the library

plflanagan reviewed (approved):
Nice!
